databroker/assets/base_registry.py

```diff
@@ -472,12 +472,14 @@ def register_resource(self, spec, root, rpath, rkwargs,
 
         col = self._resource_col
 
-        return self._api.insert_resource(col, spec, rpath,
-                                         rkwargs,
-                                         self.known_spec,
-                                         root=root,
-                                         path_semantics=path_semantics,
-                                         **rs_kwarg)['uid']
+        return self._api.insert_resource(
+            col, spec, rpath,
+            rkwargs,
+            self.known_spec,
+            root=root,
+            path_semantics=path_semantics,
+            duplicate_exc=self.DuplicateKeyError,
```

Comment:
Good catch.

Reply:
This was failing on Python master (as it was trying to do `except None` and at some point throwing / catching things that do not derive from BaseException will be forbidden).